core/src/main/scala/org/apache/spark/rdd/ShuffledRDD.scala

```diff
@@ -104,10 +105,26 @@ class ShuffledRDD[K: ClassTag, V: ClassTag, C: ClassTag](
   }
 
   override def compute(split: Partition, context: TaskContext): Iterator[(K, C)] = {
+    // Use -1 for our Shuffle ID since we are on the read side of the shuffle.
+    val shuffleWriteId = -1
+    // If our task has data property accumulators we need to keep track of which partitions
+    // we are processing.
+    if (context.taskMetrics.hasDataPropertyAccumulators()) {
+      context.setRDDPartitionInfo(id, shuffleWriteId, split.index)
+    }
     val dep = dependencies.head.asInstanceOf[ShuffleDependency[K, V, C]]
-    SparkEnv.get.shuffleManager.getReader(dep.shuffleHandle, split.index, split.index + 1, context)
+    val itr = SparkEnv.get.shuffleManager.getReader(dep.shuffleHandle, split.index, split.index + 1,
+      context)
```

Comment:
huh github formatted my response in a seperate thread trying to reposting it underneath
So unless the input was already sorted we will always have to read the entire input first. Also since we have tests for this if we do change this its a simple matter of updating the wrapper to be more like the rest should we ever change this.